tool/tsh/tsh.go

```diff
@@ -2018,20 +2018,58 @@ func reissueWithRequests(cf *CLIConf, tc *client.TeleportClient, reqIDs ...strin
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
 	params := client.ReissueParams{
 		AccessRequests: reqIDs,
 		RouteToCluster: cf.SiteName,
 	}
+
 	// if the certificate already had active requests, add them to our inputs parameters.
 	if len(profile.ActiveRequests.AccessRequests) > 0 {
 		params.AccessRequests = append(params.AccessRequests, profile.ActiveRequests.AccessRequests...)
 	}
+
 	if params.RouteToCluster == "" {
 		params.RouteToCluster = profile.Cluster
 	}
-	if err := tc.ReissueUserCerts(cf.Context, params); err != nil {
+
+	// Look up any existing key for the cluster. We will need this to check whether
+	// the granted roles have changed post-approval. For our purposes here, it's OK
+	// for this oldKey not to exist.
+
+	oldKey, err := tc.LocalAgent().GetKey(profile.Cluster, client.WithSSHCerts{})
+	switch {
+	case trace.IsNotFound(err):
+		oldKey = nil
+	case err != nil:
 		return trace.Wrap(err)
+	default:
+		// carry on
 	}
```

Comment:
```suggestion
	if err != nil && !trace.IsNotFound(err) {
		return trace.Wrap(err)
	}
```

Reply:
Never mind. I'm drawing the "old" Roles list from the profile now, so the whole thing disappears

Reply:
And the whole think gas been deleted in  any case